Author: Scott Salley <ssalley@beyondtrust.com>
Description: The gssspnego plugin was not properly separated from the cyrus-sasl
sources that Likewise/BeyondTrust builds internally. This allows it to be built
with libsasl2-dev
Forwarded: yes
Last-Update: 2011-07-28

diff -Nurb likewise-open-6.1.0.406/cyrus-sasl/plugins/gssspnego.c likewise-open-6.1.0.406.new/cyrus-sasl/plugins/gssspnego.c
--- likewise-open-6.1.0.406/cyrus-sasl/plugins/gssspnego.c	2011-07-27 12:01:44.000000000 -0700
+++ likewise-open-6.1.0.406.new/cyrus-sasl/plugins/gssspnego.c	2011-07-28 15:45:47.000000000 -0700
@@ -69,9 +69,9 @@
 #endif /* WIN32 */
 #include <fcntl.h>
 #include <stdio.h>
-#include <sasl.h>
-#include <saslutil.h>
-#include <saslplug.h>
+#include <sasl/sasl.h>
+#include <sasl/saslutil.h>
+#include <sasl/saslplug.h>
 
 #include "plugin_common.h"
 
@@ -81,6 +81,7 @@
 
 #include <errno.h>
 
+#include <string.h>
 /*****************************  Common Section  *****************************/
 
 static const char plugin_id[] = "$Id: gssapi.c,v 1.92 2004/07/21 14:39:06 rjs3 Exp $";
diff -Nurb likewise-open-6.1.0.406/cyrus-sasl/plugins/gssspnego_init.c likewise-open-6.1.0.406.new/cyrus-sasl/plugins/gssspnego_init.c
--- likewise-open-6.1.0.406/cyrus-sasl/plugins/gssspnego_init.c	2011-07-27 12:01:44.000000000 -0700
+++ likewise-open-6.1.0.406.new/cyrus-sasl/plugins/gssspnego_init.c	2011-07-28 15:54:05.000000000 -0700
@@ -10,16 +10,18 @@
 #include <fcntl.h>
 #include <assert.h>
 
-#include <sasl.h>
-#include <saslplug.h>
-#include <saslutil.h>
+#include <sasl/sasl.h>
+#include <sasl/saslplug.h>
+#include <sasl/saslutil.h>
 
 #include "plugin_common.h"
 
 #ifdef macintosh
-#include <sasl_plugin_decl.h>
+#include <sasl/sasl_plugin_decl.h>
 #endif
 
+#include <string.h>
+
 #ifdef WIN32
 BOOL APIENTRY DllMain( HANDLE hModule, 
                        DWORD  ul_reason_for_call, 
diff -Nurb likewise-open-6.1.0.406/cyrus-sasl/plugins/plugin_common.c likewise-open-6.1.0.406.new/cyrus-sasl/plugins/plugin_common.c
--- likewise-open-6.1.0.406/cyrus-sasl/plugins/plugin_common.c	2011-07-27 12:01:44.000000000 -0700
+++ likewise-open-6.1.0.406.new/cyrus-sasl/plugins/plugin_common.c	2011-07-28 15:46:14.000000000 -0700
@@ -58,9 +58,9 @@
 #include <unistd.h>
 #endif
 #include <fcntl.h>
-#include <sasl.h>
-#include <saslutil.h>
-#include <saslplug.h>
+#include <sasl/sasl.h>
+#include <sasl/saslutil.h>
+#include <sasl/saslplug.h>
 
 #include <errno.h>
 #include <ctype.h>
diff -Nurb likewise-open-6.1.0.406/cyrus-sasl/plugins/plugin_common.h likewise-open-6.1.0.406.new/cyrus-sasl/plugins/plugin_common.h
--- likewise-open-6.1.0.406/cyrus-sasl/plugins/plugin_common.h	2011-07-27 12:01:44.000000000 -0700
+++ likewise-open-6.1.0.406.new/cyrus-sasl/plugins/plugin_common.h	2011-07-28 15:46:40.000000000 -0700
@@ -59,9 +59,11 @@
 #endif /* WIN32 */
 #endif /* macintosh */
 
-#include <sasl.h>
-#include <saslutil.h>
-#include <saslplug.h>
+#include <sasl/sasl.h>
+#include <sasl/saslutil.h>
+#include <sasl/saslplug.h>
+
+#include <string.h>
 
 #ifdef WIN32
 #define PLUG_API __declspec(dllexport)
diff -Nurb likewise-open-6.1.0.406/gssspnego-cyrus-sasl/MakeKitBuild likewise-open-6.1.0.406.new/gssspnego-cyrus-sasl/MakeKitBuild
--- likewise-open-6.1.0.406/gssspnego-cyrus-sasl/MakeKitBuild	1969-12-31 16:00:00.000000000 -0800
+++ likewise-open-6.1.0.406.new/gssspnego-cyrus-sasl/MakeKitBuild	2011-07-28 16:51:33.000000000 -0700
@@ -0,0 +1,26 @@
+configure()
+{
+    mk_config_header "config.h"
+
+    lw_define_feature_macros
+
+    MK_CFLAGS="$MK_CFLAGS -Wall -fno-strict-aliasing"
+    MK_CXXFLAGS="$MK_CXXFLAGS -Wall"
+
+    mk_check_headers \
+        unistd.h gssapi.h inttypes.h
+}
+
+make()
+{
+    mk_library \
+        LIB=gssspnego \
+        SOURCES="../cyrus-sasl/plugins/gssspnego.c \
+                 ../cyrus-sasl/plugins/gssspnego_init.c \
+                 ../cyrus-sasl/plugins/plugin_common.c" \
+        CPPFLAGS="-DSASL_CB_SERVERFQDN=0xc0004001" \
+        INCLUDEDIRS=". ../cyrus-sasl/plugins" \
+        HEADERDEPS="" \
+        LIBDEPS="gssapi_krb5" \
+        INSTALLDIR="/usr/lib/sasl2"
+}
